Adding new function sg_params_checker to check acquisition parameters; Include it in CI

alexfoias opened the pull request:
This PR introduces parameters checking during continuous integration (CI), via Travis.

The CLI is also introduced: `sg_params_checker`.

### How to test this PR
~~~
git checkout -b af/acq_param_ci origin/af/acq_param_ci
pip install -e .
sg_params_checker -path-in <PATH>/data-single-subject/
~~~

Here are some results for current configuration.

https://travis-ci.com/github/spine-generic/data-single-subject/jobs/360645135#L275

Fixes #113

alexfoias commented:
Here are the results from the latest version of the implementation: https://travis-ci.com/github/spine-generic/data-single-subject

alexfoias commented:
@jcohenadad This PR is ready for review. I'll first nee to merge this one and then I can implement CI in the single & multi subject datasets.

jcohenadad commented on spinegeneric/cli/params_checker.py:
```diff
@@ -0,0 +1,82 @@
+            print(line)
+
+if __name__ == '__main__':
+    main()
```
EOF

jcohenadad commented on spinegeneric/cli/params_checker.py:
```diff
@@ -0,0 +1,82 @@
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Acquistion parameters checker",
```
add more info about how this is done.

jcohenadad commented:
this should pass:
~~~
WARNING: Incorrect RepetitionTime: sub-chibaIngenia_T2star.nii.gz; TR=0.625 instead of 0.6
WARNING: Incorrect EchoTime: sub-douglas_T2w.nii.gz; TE=0.12 instead of 0.118
WARNING: Incorrect RepetitionTime: sub-glen_T2star.nii.gz; TR=0.625 instead of 0.6
~~~

alexfoias commented:
I updated the limits and addressed the other comments. 

As for the limits here is the output:

```
(base) alfoi@MacBook-Pro:spine-generic % sg_params_checker -path-in /Users/alfoi/code/data-single-subject/
/Users/alfoi/miniconda3/lib/python3.7/site-packages/bids/layout/models.py:102: FutureWarning: The 'extension' entity currently excludes the leading dot ('.'). As of version 0.14.0, it will include the leading dot. To suppress this warning and include the leading dot, use `bids.config.set_option('extension_initial_dot', True)`.
  FutureWarning)
WARNING: Incorrect FlipAngle: sub-douglas_T2w.nii.gz; FA=120 instead of 180
WARNING: Incorrect RepetitionTime: sub-mgh_T2w.nii.gz; TR=2 instead of 1.5
WARNING: Incorrect FlipAngle: sub-tokyoSigna1_T2star.nii.gz; FA=20 instead of 30
WARNING: Incorrect FlipAngle: sub-tokyoSigna2_T2star.nii.gz; FA=20 instead of 30
WARNING:sub-ucl_T2star.nii.gz Missing Manufacturer in json sidecar; Cannot check parameters.
```

jcohenadad commented on spinegeneric/cli/params_checker.py:
```diff
@@ -1,3 +1,11 @@
+#!/usr/bin/env python
+#
+# Script to check parameters checker
```
```suggestion
# Script to check acquisition parameters.
```

jcohenadad commented on spinegeneric/cli/params_checker.py:
```diff
@@ -1,3 +1,11 @@
+#
+# Script to check parameters checker
+#
+# For usage, type: sg_params_checker -path-in PATH_TO_BIDSDATASET
```
```suggestion
# For usage, type: sg_params_checker -h
```
less maintenance

jcohenadad commented on spinegeneric/cli/params_checker.py:
```diff
@@ -11,7 +19,9 @@
-        description="Acquistion parameters checker",
+        description="Acquistion parameters checker feature. This feature allows the users"
+                    "to compare the acquisition parameters that can be found in the json "
+                    "sidecar to the recommended acquisition parameters."
```
```suggestion
        description="This function checks if the image acquisition parameters of each "
                    "image (found in the json file) in a dataset conform to the required parameters "
                    "(listed for each vendor in the file cli/specs.json)."
```

jcohenadad commented on spinegeneric/cli/params_checker.py:
```diff
@@ -80,4 +90,3 @@ def main():
 
 if __name__ == '__main__':
     main()
-    
```
EOF

alexfoias commented:
@jcohenadad Here is the output of the new version:
```
(base) alfoi@MacBook-Pro:~ % sg_params_checker -path-in /Users/alfoi/code/data-single-subject/
/Users/alfoi/miniconda3/lib/python3.7/site-packages/bids/layout/models.py:102: FutureWarning: The 'extension' entity currently excludes the leading dot ('.'). As of version 0.14.0, it will include the leading dot. To suppress this warning and include the leading dot, use `bids.config.set_option('extension_initial_dot', True)`.
  FutureWarning)
WARNING: Incorrect FlipAngle: sub-douglas_T2w.nii.gz; FA=120 instead of 180
WARNING: Incorrect RepetitionTime: sub-mgh_T2w.nii.gz; TR=2 instead of 1.5
WARNING: Incorrect FlipAngle: sub-tokyoSigna1_T2star.nii.gz; FA=20 instead of 30
WARNING: Incorrect FlipAngle: sub-tokyoSigna2_T2star.nii.gz; FA=20 instead of 30
WARNING:sub-ucl_T2star.nii.gz Missing Manufacturer in json sidecar; Cannot check parameters.
```
